app/models/user.rb

```diff
@@ -122,6 +122,9 @@ def moderatable
 
   after_save :nullify_authentication_id, if: :any_sso_authorizations?
 
+  after_destroy :update_next_account_first_admin, if: proc { account.first_admin_id == id }
```

Comment:
This seems important to the logic. How about defining a method
```ruby
def first_admin?
  account.first_admin_id == id
end
```
and change the conditional to
```ruby
.., if: :first_admin?
```

Reply:
makes sense 👍